tools/SyncProjects.ps1

```diff
@@ -0,0 +1,264 @@
+  $nextLink = $response.RelationLink.next
+  do {
+    if ($nextLink -ne $null) {
+      $response = Invoke-GithubAPI -AbsoluteUri $nextLink -Raw
+      $result = $result + ($response.Content | ConvertFrom-Json)
+      $nextLink = $response.RelationLink.next
+    }
+  }
+  while ($nextLink -ne $null)
+
+  Write-Output $result
+}
+
+Function Get-PDKProjects {
+  Invoke-GithubAPIWithPaging -RelativeUri '/repos/puppetlabs/pdk-planning/projects?state=open'
+}
+
+Function Convert-GHIssueToNote {
+  param(
+    [Parameter(Mandatory = $true, ValueFromPipeline = $true)]
+    [Object]$Issue
+  )
+
+  Process {
+    # We only add the title, because github fills in the description as a rich-link
+    $hash = @{
+      'cardId' = -1
+      'note'   = @"
+[GitHub Issue $($issue.number)]($($issue.html_url))
+
+$($issue.title)
+"@
+      'expectedColumn' = 'To do'
+    }
+    if ($Issue.state -eq 'closed') { $hash['expectedColumn'] = 'Done'} # What about in progress? (Is that a PR versus an Issue?)
+    Write-Output ([PSCustomObject]$hash)
+  }
+}
+
+Function Convert-GHCardToNote {
+  param(
+    [Parameter(Mandatory = $true, ValueFromPipeline = $true)]
+    [Object]$Card,
+
+    [String]$ColumnName
+  )
+
+  Process {
+    # Write-Host ($Card | ConvertTo-JSON) -ForegroundColor Red
+    $hash = @{
+      'foundMatchingIssue' = $false
+      'note'               = $Card.note
+      'id'                 = $Card.id
+      'currentColumn'      = $ColumnName
+      'expectedColumn'     = '????'
+    }
+    Write-Output ([PSCustomObject]$hash)
+  }
+}
+
+Function Get-JiraIssues($fixVersion) {
+  Set-JiraConfigServer $script:JiraServer
+
+  $Query = "project = `"Puppet Development Kit`" and fixVersion = `"${fixVersion}`""
+
+  try {
+    Get-JiraIssue -Query $Query -Fields "-comment" -ErrorAction 'Stop' | ForEach-Object {
+      $Issue = $_
+
+      if ($issue.Description.Length -gt 100) {
+        $description = $issue.Description.SubString(0, 100) + "..."
+      }
+      else {
+        $description = $issue.Description
+      }
+
+      $hash = @{
+        'cardId' = -1
+        'note'           = @"
+[Jira Issue $($Issue.key)]($($issue.HttpUrl))
+
+$($issue.Summary)
+
+---
+
+${description}
+"@
+        'expectedColumn' = 'To do'
+      }
+      if ($Issue.state -eq 'Closed') { $hash['expectedColumn'] = 'Done'}
+      if ($Issue.state -eq 'Resolved') { $hash['expectedColumn'] = 'Done'}
+      if ($Issue.state -eq 'Completed') { $hash['expectedColumn'] = 'Done'}
+      Write-Output ([PSCustomObject]$hash)
+    }
+  } catch {
+    Write-Warning "Error querying Jira $_"
+  }
+}
+
+Function Invoke-ParsePDKProject($project) {
+  if ($project.name -notmatch '^Release ([0-9.]+)$') {
+    Write-Verbose "Project $($project.name) is not a release"
+    return
+  }
+  Write-Verbose "Parsing Project $($project.name) ..."
+  $ProjectVersion = $Matches[1]
+  $MilestoneName = $ProjectVersion
+  $JiraFixVersion = $ProjectVersion
```

Comment:
We have been using the format "PDK x.y.z" for the fix versions in JIRA thus far: https://tickets.puppetlabs.com/projects/PDK?selectedItem=com.atlassian.jira.jira-projects-plugin%3Arelease-page&status=released-unreleased